src/feature/dirauth/process_descs.c

```diff
@@ -238,11 +238,16 @@ dirserv_load_fingerprint_file(void)
       is_valid_fpr = false;
     }
 
-    tor_assert(is_ed25519 || is_rsa);
+    if (!is_ed25519 || !is_rsa) {
```

Comment:
This will trigger every time literally. Perhaps just use the `if` check above to issue the log msg? Also maybe turn it to `log_warn` and also please change the log text body so that it's not identical to the one below.